projects/js-packages/publicize-components/src/components/share-status/retry.tsx

```diff
@@ -0,0 +1,82 @@
+import { IconTooltip } from '@automattic/jetpack-components';
+import { useAnalytics } from '@automattic/jetpack-shared-extension-utils';
+import { Button } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { __, _x } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import useSharePost from '../../hooks/use-share-post';
+import { store as socialStore } from '../../social-store';
+import { ShareStatusItem } from '../../social-store/types';
+import { connectionMatchesShareItem } from '../../utils/share-status';
+import styles from './styles.module.scss';
+
+export type RetryProps = {
+	shareItem: ShareStatusItem;
+};
+
+/**
+ * Retry component.
+ *
+ * @param {RetryProps} props - component props
+ *
+ * @return {import('react').ReactNode} - React element
+ */
+export function Retry( { shareItem }: RetryProps ) {
```

Comment:
I extracted this to its own component because there is more logic going to be added to it for refreshing the data when polling.